src/Storages/StorageFuzzJSON.cpp

```diff
@@ -0,0 +1,511 @@
+#include <optional>
+#include <random>
+#include <Columns/ColumnString.h>
+#include <Storages/StorageFactory.h>
+#include <Storages/StorageFuzzJSON.h>
+#include <Storages/checkAndGetLiteralArgument.h>
+#include <Common/JSONParsers/RapidJSONParser.h>
+#include <Common/JSONParsers/SimdJSONParser.h>
+#include <Common/checkStackSize.h>
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+
+extern const int NUMBER_OF_ARGUMENTS_DOESNT_MATCH;
+extern const int NOT_IMPLEMENTED;
+extern const int INCORRECT_DATA;
+}
+
+using uniform = std::uniform_int_distribution<size_t>;
+using Node = StorageFuzzJSON::JSONNode;
+using JSONNodeList = std::list<std::shared_ptr<Node>>;
+
+using FuzzConfig = StorageFuzzJSON::Config;
+
+struct StorageFuzzJSON::JSONNode
+{
+    enum class ValueType : size_t
+    {
+        Fixed = 0,
+        Array = 1,
+        Nested = 2,
+    };
+
+    static ValueType getValueType(const JSONNode & n)
+    {
+        if (n.fixed)
+            return ValueType::Fixed;
+        else if (n.array)
+            return ValueType::Array;
+        else if (n.nested)
+            return ValueType::Nested;
+        else
+            throw Exception(ErrorCodes::NOT_IMPLEMENTED, "Unknown value type");
+    }
+
+    // A node can represent a JSON value; then it does not have a key.
+    std::optional<String> key;
+
+    // The node value must be one of the following:
+    std::optional<Field> fixed;
+    std::optional<JSONNodeList> array;
+    std::optional<JSONNodeList> nested;
```

Comment:
Why not to store keys in parent node, btw? To have a map in nested field

```suggestion
    // A node can represent a JSON value; then it does not have a key.
    // It set only if this node is child of any other node and stored in `nested` of that node
    std::optional<String> key;

    // The node value must be one of the following:
    // Examples: 5, true, "abc"
    std::optional<Field> fixed;
    // Examples: [], ["a"], [1, true]
    std::optional<JSONNodeList> array;
    // Examples: {}, {"a": [1,2], "b": "c"}
    std::optional<JSONNodeList> nested;
```

Reply:
> Why not to store keys in parent node, btw? To have a map in nested field

There is no use case for a quick lookup by a key, also map would introduce key reordering.
In the context of this diff, a 'node' is either a key-value pair or a value.